crates/cheatcodes/src/inspector.rs

```diff
@@ -1740,13 +1758,15 @@ where {
                         ecx_inner.journaled_state.state().get_mut(&broadcast.new_origin).unwrap();
 
                     let zk_tx = if self.use_zk_vm {
-                        let injected_factory_deps = self.zk_use_factory_deps.iter().map(|contract| {
-                            crate::fs::get_artifact_code(self, contract, false)
+                        let injected_factory_deps = self.zk_use_factory_deps.iter().flat_map(|contract| {
```

Comment:
since there's some duplicaton here, maybe there's room for extracting things on a method that does both the fetching and the clearing of the factory deps in one go